### PR TITLE
use pdfFillReducer to fill BerH pdf

### PR DESCRIPTION
**Pull request by rbrtrfl**
- move fillOutFunction one level up
- use pdfFillReducer to fill BerH pdf

### CODE REVIEW

File: app/services/pdf/beratungshilfe/sections/F_besitz/F_besitz.ts [hunk 1]
```diff
@@ -1,107 +1,116 @@
...
-  attachment: AttachmentEntries,
-  pdfFields: BeratungshilfePDF,
-  context: BeratungshilfeFormularContext,
-) {
-  const { hasBankkonto, bankkonten, eigentumTotalWorth } = context;
+  return {
+    pdfValues,
+    attachment,
+  };
+};
```
Comment (Spencer6497) [inline app/services/pdf/beratungshilfe/sections/F_besitz/F_besitz.ts:38]: we should be destructuring and returning the result of `pdfFillReducer`, like so:
```
const { pdfValues, attachment } = pdfFillReducer(...)
return {
    pdfValues,
    attachment
}
```

does the way it currently is work?

File: app/services/pdf/beratungshilfe/sections/G_ausgaben.ts [hunk 2]
```diff
@@ -25,53 +26,52 @@ export const ausgabenSituationMapping = {
...
-) {
-  const ausgaben = context.ausgaben ?? [];
+export const fillAusgaben: BerHPdfFillFunction = ({ userData, pdfValues }) => {
+  const attachment: AttachmentEntries = [];
+
+  const ausgaben = userData.ausgaben ?? [];
   const hasOverflowAusgaben = ausgaben.length > AUSGABEN_MAX_COUNT_FIELDS;
 
-  const isPdfFieldExceedsMaxChars = context.ausgaben?.some(
+  const isPdfFieldExceedsMaxChars = userData.ausgaben?.some(
```
Comment (Spencer6497) [inline app/services/pdf/beratungshilfe/sections/G_ausgaben.ts:35]:
```suggestion
  const isPdfFieldExceedsMaxChars = ausgaben.some(
```

**Review by Spencer6497 — CHANGES_REQUESTED** [review]
Great work on this 🙌 though I'm fairly certain there's a logical error where noted ⬇️

File: app/services/pdf/beratungshilfe/sections/F_besitz/F_besitz.ts (hunk 1 above)
Comment (rbrtrfl) [reply app/services/pdf/beratungshilfe/sections/F_besitz/F_besitz.ts]: you are absolutely right!

File: app/services/pdf/beratungshilfe/sections/A_angelegenheit.ts [hunk 3]
```diff
@@ -20,20 +20,20 @@ const bereichMapping = {
...
 
-export function fillAngelegenheit(
-  attachment: AttachmentEntries,
-  pdfFields: BeratungshilfePDF,
-  context: BeratungshilfeFormularContext,
-) {
+export const fillAngelegenheit: BerHPdfFillFunction = ({
+  userData,
+  pdfValues,
+}) => {
```
[the remark this answers is not in the document]
Comment (rbrtrfl) [reply app/services/pdf/beratungshilfe/sections/A_angelegenheit.ts:26]: As far as I understand, a type alias like `BerHPdfFillFunction` can only be used like this with anonymous function definitions. If we would go for a named function, explicit typing for parameters and return value would be necessary.

**Review by Spencer6497 — APPROVED** [review]

**Review by chohner — APPROVED** [review]
lgtm

**Pull request merged**